genesyscloud/resource_genesyscloud_media_retention_policy.go

```diff
@@ -0,0 +1,5258 @@
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice([]string{"NONE", "BEST", "ALL", ""}, false),
+			},
+			"queue_flow": {
+				Description: "The in-queue flow to use for call conversations waiting in queue.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Elem:        domainEntityRef,
+			},
+			"whisper_prompt": {
+				Description: "The prompt used for whisper on the queue, if configured.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Elem:        domainEntityRef,
+			},
+			"auto_answer_only": {
+				Description: "Specifies whether the configured whisper should play for all ACD calls, or only for those which are auto-answered.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"enable_transcription": {
+				Description: "Indicates whether voice transcription is enabled for this queue.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+			},
+			"enable_manual_assignment": {
+				Description: "Indicates whether manual assignment is enabled for this queue.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+			},
+			"calling_party_name": {
+				Description: "The name to use for caller identification for outbound calls from this queue.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"calling_party_number": {
+				Description: "The phone number to use for caller identification for outbound calls from this queue.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"outbound_messaging_addresses": {
+				Description: "The messaging addresses for the queue.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Elem:        queueMessagingAddresses,
+			},
+			"outbound_email_address": {
+				Description: "The messaging addresses for the queue.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Elem:        queueEmailAddress,
+			},
+		},
+	}
+
+	publishedSurveyFormReference = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"context_id": {
+				Description: "The context id of this form.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+	}
+
+	domainEntityRef = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"name": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"self_uri": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+	}
+
+	// This schema var causes an InvalidInitCycle exception
```

Comment:
Left these commented out fields in for visibility as to why they aren't present.  They contain circular references which cause the exception in terraform.  I can remove if it's called for.